ruby/ql/lib/codeql/ruby/ast/internal/Literal.qll

```diff
@@ -3,6 +3,7 @@ private import AST
 private import Constant
 private import TreeSitter
 private import codeql.ruby.controlflow.CfgNodes
+private import codeql.NumberUtils
 
 int parseInteger(Ruby::Integer i) {
```

Comment:
The implementation of `parseInteger` should probably use `NumberUtils`

Reply:
OK, I'll have a go at that.

Reply:
I've done this locally, but my implementation is more conservative. My new `parseHexInt` returns no value for `0xdeadbeef`, while the current `parseInteger` implementation returns `-559038737` (i.e. wraps to a negative number). Ruby treats `0xdeadbeef` as the positive/unsigned integer `3735928559`.

Should I attempt to update my implementation of `parseHexInt` to allow wrapping, even if it's not correct?

Reply:
There is no problem changing the meaning of `parseInteger` if the old definition is incorrect. Returning no value is probably better than incorrect wrapping.